luda-editor/client/src/app/login.rs

```diff
@@ -51,35 +39,61 @@ impl App {
 
 pub fn check_token() {
     namui::spawn_local(async {
-        match namui::cache::get("AccessToken").await.unwrap() {
+        match namui::cache::get("SessionId").await.unwrap() {
             Some(token) => {
-                namui::event::send(Event::AccessToken(String::from_utf8(token.into()).unwrap()));
-            }
-            None => {
-                let url = format!("https://github.com/login/oauth/authorize?client_id={CLIENT_ID}&scope=public_repo");
-                namui::open_external(url.as_str());
+                namui::event::send(Event::SessionId(String::from_utf8(token.into()).unwrap()));
             }
+            None => match request_github_auth_code().await {
+                Ok(code) => login_with_oauth_code(code),
+                Err(error) => {
+                    namui::event::send(Event::Error(error.to_string()));
+                }
+            },
         }
     });
 }
+
+async fn request_github_auth_code() -> Result<String, Box<dyn std::error::Error>> {
+    let client_id = CLIENT_ID.unwrap_or(DEV_CLIENT_ID);
+    let redirect_uri = web_sys::window().unwrap().location().href().unwrap();
+    let url = format!("https://github.com/login/oauth/authorize?client_id={client_id}&redirect_uri=https://sslwiheugl5ojmqlecerzhng740cekqc.lambda-url.ap-northeast-2.on.aws/{redirect_uri}");
```

Comment:
FYI: https://sslwiheugl5ojmqlecerzhng740cekqc.lambda-url.ap-northeast-2.on.aws/{redirect_uri} is our redirector url. It's on our company AWS account's Lambda.

Reply:
This is for multiple oauth callback. github only support one callback, so I put redirector for local and test server.